test/unit/transaction.test.js

```diff
@@ -23,1251 +16,1326 @@ const Segment = require('../../lib/transaction/trace/segment')
-    }).throws(/must be bound to the agent/)
-  })
-
-  it('should create a trace on demand', function () {
-    const trace = trans.trace
-    expect(trace).instanceOf(Trace)
-    expect(trans.trace).equal(trace)
-  })
-
-  it('should have at most one associated trace', function () {
-    const trace = trans.trace
-    expect(trace).not.instanceof(Array)
-  })
+  t.test('basic transaction tests', (t) => {
+    t.throws(
+      () => {
+        return new Transaction()
+      },
+      /must be bound to the agent/,
+      'should require an agent to create new transactions'
+    )
 
-  it('should hand its metrics off to the agent upon finalization', function (done) {
-    agent.on('transactionFinished', function (inner) {
-      expect(inner.metrics).equal(trans.metrics)
+    const trace = txn.trace
+    t.ok(trace instanceof Trace, 'should create a trace on demand')
+    t.notOk(trace instanceof Array, 'should have at most one associated trace')
 
-      return done()
+    agent.on('transactionFinished', (inner) => {
+      t.equal(
+        inner.metrics,
+        txn.metrics,
+        'should hand its metrics off to the agent upon finalization'
+      )
+      t.end()
     })
 
-    trans.end()
+    txn.end()
   })
 
-  describe('when distributed tracing is enabled', function () {
-    beforeEach(function () {
-      agent.config.distributed_tracing.enabled = true
-    })
+  t.test('with distributed tracing enabled', (t) => {
+    agent.config.distributed_tracing.enabled = true
 
-    afterEach(function () {
-      agent.config.distributed_tracing.enabled = false
+    agent.once('transactionFinished', () => {
+      t.equal(agent.spanEventAggregator.length, 1, 'should produce span events when finalizing')
     })
-
-    it('should produce span events when finalizing', function (done) {
-      agent.once('transactionFinished', function () {
-        expect(agent.spanEventAggregator.length).to.equal(1)
-
-        return done()
-      })
-      helper.runInTransaction(agent, function (txn) {
-        const childSegment = txn.trace.add('child')
-        childSegment.start()
-
-        txn.end()
-      })
+    helper.runInTransaction(agent, function (inner) {
+      const childSegment = inner.trace.add('child')
+      childSegment.start()
+      inner.end()
     })
 
-    it('should not produce span events when ignored', function (done) {
-      agent.once('transactionFinished', function () {
-        expect(agent.spanEventAggregator.length).to.equal(0)
-
-        return done()
-      })
-
-      helper.runInTransaction(agent, function (txn) {
-        const childSegment = txn.trace.add('child')
-        childSegment.start()
-
-        txn.ignore = true
-        txn.end()
-      })
-    })
+    t.end()
   })
 
-  it('should hand itself off to the agent upon finalization', function (done) {
-    agent.on('transactionFinished', function (inner) {
-      expect(inner).equal(trans)
+  t.test('with distributed tracing enabled', (t) => {
```

Comment:
Is this the same as the block on line 59?

Reply:
It's not, on line 59 we are not ignoring the transaction. I've relabelled the tests to make this clearer.